src/slider/slider.component.ts

```diff
@@ -0,0 +1,328 @@
+	registerOnChange(fn: any) {
+		this.propagateChange = fn;
+	}
+
+	/** Callback to notify the model when our input has been touched */
+	onTouched: () => any = () => { };
+
+	/** Register a callback to notify when our input has been touched */
+	registerOnTouched(fn: any) {
+		this.onTouched = fn;
+	}
+
+	/** Receives a value from the model */
+	writeValue(v: any) {
+		this.value = v;
+	}
+
+	/** Returns the amount of "completeness" as a fraction of the total track width */
+	getPercentComplete() {
+		const trackWidth = this.track.nativeElement.getBoundingClientRect().width;
+		return this.slidAmount / trackWidth;
+	}
+
+	/** Helper function to return the CSS transform `scaleX` function */
+	scaleX(complete) {
+		return `scaleX(${complete})`;
+	}
+
+	/** Converts a given px value to a "real" value in our range */
+	convertToValue(pxAmount) {
+		// basic concept borrowed from carbon-components
+		// ref: https://github.com/IBM/carbon-components/blob/43bf3abdc2f8bdaa38aa84e0f733adde1e1e8894/src/components/slider/slider.js#L147-L151
+		const range = this.max - this.min;
+		const trackWidth = this.track.nativeElement.getBoundingClientRect().width;
+		const unrounded = pxAmount / trackWidth;
+		const rounded = Math.round((range * unrounded) / this.step) * this.step;
+		return rounded + this.min;
+	}
+
+	/** Converts a given "real" value to a px value we can update the view with */
+	convertToPx(value) {
+		const trackWidth = this.track.nativeElement.getBoundingClientRect().width;
+		if (value >= this.max) {
+			return trackWidth;
+		}
+
+		if (value <= this.min) {
+			return 0;
+		}
+
+		return Math.round(trackWidth * (value / this.max));
+	}
+
+	/**
+	 * Increments the value by the step value, or the step value multiplied by the `multiplier` argument.
+	 *
+	 * @argument multiplier Defaults to `1`, multiplied with the step value.
+	 */
+	incrementValue(multiplier = 1) {
+		this.value = this.value + (this.step * multiplier);
+	}
+
+	/**
+	 * Decrements the value by the step value, or the step value multiplied by the `multiplier` argument.
+	 *
+	 * @argument multiplier Defaults to `1`, multiplied with the step value.
+	 */
+	decrementValue(multiplier = 1) {
+		this.value = this.value - (this.step * multiplier);
+	}
+
+	/** Change handler for the optional input */
+	onChange(event) {
+		this.value = event.target.value;
+	}
+
+	/** Handles clicks on the range track, and setting the value to it's "real" equivalent */
+	onClick(event) {
+		if (this.disabled) { return; }
+		const trackLeft = this.track.nativeElement.getBoundingClientRect().left;
+		this.value = this.convertToValue(event.clientX - trackLeft);
+		console.log(event);
+	}
+
+	/** Focus handler for the optional input */
+	onFocus({target}) {
+		target.select();
+	}
+
+	/** Mouse move handler. Responsible for updating the value and visual selection based on mouse movement */
+	onMouseMove(event) {
+		if (this.disabled) { return; }
+		if (this.isMouseDown) {
```

Comment:
This could also be a guard (like for disabled), indicating that function does nothing if mouse isn't down.
It could be combined withing the same if statement with disabled.